### PR TITLE
Consume ValueTask with GetAwaiter().GetResult()

### PR DESCRIPTION
**Pull request by Kahbazi**

### CODE REVIEW

File: src/Components/Components/src/Rendering/ComponentState.cs [hunk 1]
```diff
@@ -236,6 +236,9 @@ public Task DisposeInBatchAsync(RenderBatchBuilder batchBuilder)
                 var result = ((IAsyncDisposable)Component).DisposeAsync();
                 if (result.IsCompletedSuccessfully)
                 {
+                    // If it's a IValueTaskSource backed ValueTask,
+                    // inform it its result has been read so it can reset
+                    result.GetAwaiter().GetResult();
```
Comment (javiercn) [inline src/Components/Components/src/Rendering/ComponentState.cs:241]: @davidfowl do we need to do this across all places where we consume ValueTask?

Comment (davidfowl) [reply src/Components/Components/src/Rendering/ComponentState.cs:241]: Yes. There's an uber issue with a bunch of violations in our code base wrt value task. @stephentoub filed the issue in 5.0 we should address those things in 6.0

Comment (Pilchie) [reply src/Components/Components/src/Rendering/ComponentState.cs:241]: Looks like it's #16876.  Might as well go ahead and take this if someone wants to sign off on it.

**Review by javiercn — APPROVED** [review]
Signing-off on the component bits.

**Review by davidfowl — APPROVED** [review]

**Comment by Pilchie** [comment]
Thanks @Kahbazi !

**Pull request merged**